### DISCUSSION TITLE
Clean interaciton with app dispatchers (rofi, wofi, dmenu, etc.)

### DISCUSSION THREAD

**Discussion by OhKayBet** [igo95862/bubblejail#116] [Q&A] [answered]
Has anyone figured out how to get bubblejail apps to cleanly interact with app launcher w/o overpopulating it with the orriginal app and  app bubble versions. all ive though is to edit the desktop version to exec bublejail run app, but that doesnt seem like a safe way of doing it.

**Comment by igo95862** [comment, marked as answer]
The way it works right now.

If you use a profile which has the desktop entry name defined then a desktop entry with the same name will be created in `~/.local/share/applications/` which will shadow the desktop entry installed on the system. For example, if you create an instance with `firefox` profile a `~/.local/share/applications/firefox.desktop` will be created and the original firefox desktop entry will no longer be accessible.

If profile does not have a desktop entry then a generic `bubble_{instance_name}.desktop` entry will be created.

If you pass `--no-desktop-entry` option then no desktop entry will be generated.

There is also a `generate-desktop-entry` command which can generate a desktop entry for an existing instance and can use any desktop entry as a template. [See Arch Wiki for an example.](https://wiki.archlinux.org/title/Bubblejail#Re-using_profile_for_similar_applications)

**Comment by OhKayBet** [comment]
appreciate it, didnt realize if the desktop name is the same itll use the user desktop hide the global desktop
